src/client/protocol.c

```diff
@@ -14,67 +14,258 @@
 #include "../tracing.h"
 #include "../tuple.h"
 
+/* Convert a value that potentially borrows data from the client_proto read buffer
+ * into one that owns its data. The owned data must be free with freeOwnedValue. */
+static int makeValueOwned(struct value *val)
+{
+	char *p;
+	switch (val->type) {
+		case SQLITE_TEXT:
+			p = strdup(val->text);
+			if (p == NULL) {
+				return DQLITE_CLIENT_PROTO_ERROR;
+			}
+			val->text = p;
+			break;
+		case DQLITE_ISO8601:
+			p = strdup(val->iso8601);
+			if (p == NULL) {
+				return DQLITE_CLIENT_PROTO_ERROR;
+			}
+			val->iso8601 = p;
+			break;
+		case SQLITE_BLOB:
+			p = malloc(val->blob.len);
```

Comment:
Just letting up a balloon, an opinion from a personal hero ;-) https://twitter.com/antirez/status/1619369355833405441?s=20&t=nG9lOE2osQLVHC6PtV2E-Q . Not saying we should do anything like that right now.